src/storage/StorageConfig.js

```diff
@@ -0,0 +1,91 @@
+const fetch = require('node-fetch')
+
+const logger = require('../helpers/logger')('streamr:storage:StorageConfig')
+
+/*
+ * Connects to Core API and queries the configuration there.
+ * Refreshes the config at regular intervals.
+ */
+const getStreamFromKey = (key) => {
+    const [id, partitionStr] = key.split('::')
+    return {
+        id,
+        partition: Number(partitionStr)
+    }
+}
+
+const getKeyFromStream = (streamId, streamPartition) => {
+    return `${streamId}::${streamPartition}`
+}
+
+module.exports = class StorageConfig {
+    // use createInstance method instead: it fetches the up-to-date config from API
+    constructor(nodeId, apiUrl) {
+        this.streams = new Set()
+        this.listeners = []
+        this.nodeId = nodeId
+        this.apiUrl = apiUrl
+        this._poller = undefined
+        this._stopPoller = false
+    }
+
+    static async createInstance(nodeId, apiUrl, pollInterval = 10 * 60 * 1000) {
+        const instance = new StorageConfig(nodeId, apiUrl)
+        const poll = async () => {
+            try {
+                await instance.refresh()
+            } catch (e) {
+                logger.warn(`Unable to refresh storage config: ${e}`)
+            }
+            // eslint-disable-next-line no-underscore-dangle
+            if (!instance._stopPoller) {
+                // eslint-disable-next-line require-atomic-updates, no-underscore-dangle
+                instance._poller = setTimeout(poll, pollInterval)
+            }
+        }
+        await poll()
+        return instance
+    }
+
+    getStreams() {
+        return Array.from(this.streams.values()).map((key) => getStreamFromKey(key))
+    }
+
+    addChangeListener(listener) {
+        this.listeners.push(listener)
+    }
+
+    _setStreams(streamKeys) {
+        const oldKeys = this.streams
+        const newKeys = new Set(streamKeys)
+        const added = new Set([...newKeys].filter((x) => !oldKeys.has(x)))
+        const removed = new Set([...oldKeys].filter((x) => !newKeys.has(x)))
+        this.streams = newKeys
+        this.listeners.forEach((listener) => {
+            added.forEach((key) => listener.onStreamAdded(getStreamFromKey(key)))
+            removed.forEach((key) => listener.onStreamRemoved(getStreamFromKey(key)))
+        })
+    }
+
+    refresh() {
```

Comment:
I'd move the definition of `refresh` before `_setStreams` for readability (top-to-bottom journey through what the class does).

Reply:
Fixed